core/src/main/java/org/neo4j/ogm/cypher/Filter.java

```diff
@@ -429,4 +449,21 @@ private String cypherPrefix(boolean addWhereClause) {
     private String negate(String expression) {
         return String.format("NOT(%s) ", expression);
     }
+
+    /**
+     * Internal class for modifying an EQUALS comparison to ignore the case of both attribute and parameter.
+     */
+    static final class CaseInsensitiveEqualsComparison extends PropertyComparison {
+        private CaseInsensitiveEqualsComparison(final PropertyComparison propertyComparison) {
+            super(propertyComparison.getValue());
+            super.setFilter(propertyComparison.getFilter());
+        }
+
+        @Override
+        public String expression(final String nodeIdentifier) {
+            final Filter filter = this.getFilter();
+            return String.format("toLower(%s.`%s`) %s toLower({ `%s` }) ", nodeIdentifier, filter.getPropertyName(),
+                filter.getComparisonOperator().getValue(), filter.uniqueParameterName());
```

Comment:
`filter.getComparisonOperator().getValue()` could be set to `ComparisonOperator.EQUALS` because it is the only supported operator. Just the parameter not the string template placeholder.